tasktiger/worker.py

```diff
@@ -23,6 +24,7 @@
 from .timeouts import JobTimeoutException
 
 LOCK_REDIS_KEY = "qslock"
+REDIS_GLOB_CHARACTER_PATTERN = re.compile(r"([?*\[\]])")
```

Comment:
The relevant characters to be escaped are `*`, `?`, `[`, and `\` (https://github.com/redis/redis/blob/155acef51ac7826ed294a4b61d891f1c7a9a40ac/src/util.c#L58). You additionally have `]` (which is fine but irrelevant), but you should also escape `\`, so this should be `re.compile(r"([?*\[\]\\])")` (or `re.compile(r"([?*\[\\])")`).

Reply:
I don't know how I overlooked the `\` character, thanks!

I am aware that it is not necessary to escape `]`, but to my knowledge Redis docs do not mention that fact and I prefer to avoid depending on undocumented behaviour.

Reply:
Which docs btw? I was trying to find Redis docs on globs but couldn't find any, so then I just looked at the source code.

Reply:
This has some examples https://redis.io/commands/keys/ but doesn't seem like a formal spec